### PR TITLE
add the getServerSideProps function and delete the getBookingStatusTypes

### PR DESCRIPTION
**Pull request by GabrielMaga1**
Hi,

I have made the required task(Move "get-booking-status-type" to Server Side Props) in this branch and I would like to merge these changes into the main branch. Could you please review my changes and let me know if they look good to you?

### CODE REVIEW

File: pages/booking/list.tsx [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
 import getMorePageData from '../../lib/get-more-page-data'
+import prisma from '../../lib/prisma'
 
 import { useEffect, useState } from 'react'
 
-function BookingList() {
+export async function getServerSideProps() {
+  const statusTypes = await prisma.booking_status.findMany()
+  return { 
+    props: {
```
Comment (keepcoding-tech) [inline pages/booking/list.tsx:12]: I don't think you need this field

File: pages/booking/list.tsx [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+import prisma from '../../lib/prisma'
 
 import { useEffect, useState } from 'react'
 
-function BookingList() {
+export async function getServerSideProps() {
+  const statusTypes = await prisma.booking_status.findMany()
+  return { 
+    props: { 
+      statusTypesData: JSON.parse(JSON.stringify(statusTypes))
```
Comment (keepcoding-tech) [inline pages/booking/list.tsx]: Can you find a better name, something like: `bookingStatusTypes`

File: pages/booking/list.tsx [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
+export async function getServerSideProps() {
+  const statusTypes = await prisma.booking_status.findMany()
+  return { 
+    props: { 
+      statusTypesData: JSON.parse(JSON.stringify(statusTypes)) 
+    } 
+  };
+}
+
+function BookingList({statusTypesData}) {
```
Comment (keepcoding-tech) [inline pages/booking/list.tsx]: Put a space after `{` and before `}` for the parameter "statusTypesData"

**Review by keepcoding-tech — CHANGES_REQUESTED** [review]

File: pages/booking/list.tsx (hunk 1 above)
Comment (GabrielMaga1) [reply pages/booking/list.tsx:12]: If I do "props:JSON.parse(JSON.stringify(statusTypes))" I get this error - SerializableError: Error serializing props returned from `getServerSideProps` in "/booking/list".
Reason: Props must be returned as a plain object from getServerSideProps: `{ props: { ... } }` (received: `[object Array]`).
    at Object.isSerializableProps (C:\Users\SAV\Desktop\SalesWell\SalesWell\node_modules\next\dist\lib\is-serializable-props.js:16:15)

**Comment by keepcoding-tech** [comment]
Ok, now you also need to delete the file at `pages/api/booking/get-booking-status-types.tsx`

**Review by S7012MY — APPROVED** [review]

**Review by keepcoding-tech — APPROVED** [review]

**Pull request merged**